src/bundles/ipfs-provider.js

```diff
@@ -104,33 +104,49 @@ const readAPIAddressSetting = () => {
   return setting == null ? null : asAPIAddress(setting)
 }
 
-const asAPIAddress = (value) => asMultiaddress(value) || asURL(value)
+const asAPIAddress = (value) => asHttpClientOptions(value) || asMultiaddress(value) || asURL(value)
 
 /**
- * Attempts to turn cast given value into `URL` instance. Return either `URL`
- * instance or `null`.
+ * Attempts to turn cast given value into URL.
+ * Return either string instance or `null`.
  * @param {any} value
  * @returns {string|null}
  */
 const asURL = (value) => {
+  if (!value || typeof value !== 'string') return null
   try {
     return new URL(value).toString()
-  } catch (_) {
-    return null
-  }
+  } catch (_) {}
+  return null
 }
 
 /**
- * Attempts to turn cast given value into `URL` instance. Return either `URL`
- * instance or `null`.
+ * Attempts to turn cast given value into Multiaddr.
+ * Return either string instance or `null`.
  */
 const asMultiaddress = (value) => {
-  if (value != null) {
-    try {
-      return multiaddr(value).toString()
-    } catch (_) {}
-  }
+  if (!value) return null
```

Comment:
NIT: I'm not fan of `!value` as it's too ambiguous (in terms of conveying intent) and requires knowledge of some of the JS quirks to fully understand e.g. `!''` is `true` and I often wonder if author intended that or simply did not consider.